protocol/tests/fileio.rs

```diff
@@ -0,0 +1,44 @@
+//  Copyright (c) Facebook, Inc. and its affiliates.
+//  SPDX-License-Identifier: Apache-2.0
+
+extern crate protocol;
+use protocol::fileio::*;
+use std::{
+    collections::HashMap,
+    path::Path,
+    sync::{Arc, RwLock},
+};
+
+#[test]
+fn test_load_data() {
+    let t = KeyedCSV {
+        headers : vec![],
+        records : HashMap::from([
+            (String::from("a"), vec![String::from("1"), String::from("10")]),
+            (String::from("b"), vec![String::from("2"), String::from("20")]),
+            (String::from("c"), vec![String::from("3"), String::from("30")]),
+            (String::from("d"), vec![String::from("4"), String::from("40")]),
+        ]),
+    };
+
+
+    let plain_data = Arc::new(RwLock::new(KeyedCSV::default()));
+    let c = Arc::clone(&plain_data);
+    load_data(plain_data, "./tests/keyed_ints.csv", false);
+
+    let r1 = c.read().unwrap();
+    assert_eq!(*r1.headers, t.headers);
+
+    let mut k1: Vec<String>= (*r1.records.keys().map(|s| s.clone()).collect::<Vec<String>>()).to_vec().clone();
+    let mut k2: Vec<String> = t.records.keys().cloned().collect();
+    k1.sort();
+    k2.sort();
+    assert_eq!(k1, k2);
+    for k in k1 {
```

Comment:
Can you `.sort()` in lines 38 and 40 so that both does not need to be `mut`?

Reply:
> Can you `.sort()` in lines 38 and 40 so that both does not need to be `mut`?

we cannot use sort for v1 since we only have a reference from RWlock.

Reply:
for v2, since get() in hashmap only return a reference, we still need to clone then sort. How can you don't use mut?